pkg/util/template/resolver.go

```diff
@@ -30,7 +30,7 @@ func (r *Resolver) ResolveHTML(desc *HTML, preferredLanguages []string) (*htmlte
 		return nil, err
 	}
 
-	return resrc.(*htmltemplate.Template), nil
+	return resrc.(*HTMLTemplatesEffectiveResource).Template, nil
```

Comment:
This commit has to be rewritten. We have to keep return []byte because all existing consumers expect []byte. You can only change EffectiveResource to return something other than *text.Template or *html.Template.

To do so, you need to audit all existing places that perform the cast. `.(\*.*Template)`

Reply:
```sh
$ grep -rni '\.(*.\+\.Template)' pkg
pkg/util/template/resolver.go:33:       return resrc.(*htmltemplate.Template), nil
pkg/util/template/resolver.go:46:       return resrc.(*htmltemplate.Template), nil
pkg/util/template/resolver.go:59:       return resrc.(*texttemplate.Template), nil
pkg/util/template/resolver.go:72:       return resrc.(*texttemplate.Template), nil
pkg/util/template/resource_test.go:42:                  tpl := result.(*htmltemplate.Template)
```